lib/utils.js

```diff
@@ -16,253 +15,286 @@ const complements = {
-  if (!isObject(target)) return b;
-  if (!isObject(b)) return target;
-  for (let key of Object.keys(b)) {
-    let desc = Object.getOwnPropertyDescriptor(b, key);
-    if (desc.hasOwnProperty('value')) {
-      if (target.hasOwnProperty(key) && isObject(desc.value)) {
-        let existing = Object.getOwnPropertyDescriptor(target, key);
-        if (isObject(existing.value) && existing.value !== desc.value) {
-          target[key] = exports.merge({}, target[key], b[key]);
+  function mixin(target, b) {
+    if (!isObject(target)) return b;
+    if (!isObject(b)) return target;
+    for (let key of Object.keys(b)) {
+      let desc = Object.getOwnPropertyDescriptor(b, key);
+      if (Object.prototype.hasOwnProperty.call(desc, 'value')) {
+        if (Object.prototype.hasOwnProperty.call(target, key) && isObject(desc.value)) {
+          let existing = Object.getOwnPropertyDescriptor(target, key);
+          if (isObject(existing.value) && existing.value !== desc.value) {
+            target[key] = merge({}, target[key], b[key]);
+          } else {
+            Reflect.defineProperty(target, key, desc);
+          }
         } else {
           Reflect.defineProperty(target, key, desc);
         }
       } else {
         Reflect.defineProperty(target, key, desc);
       }
-    } else {
-      Reflect.defineProperty(target, key, desc);
     }
+    return target;
   }
-  return target;
-};
 
-exports.merge = (...args) => {
-  let target = {};
-  for (let ele of args) exports.mixin(target, ele);
-  return target;
-};
+  function merge(...args) {
+    let target = {};
+    for (let ele of args) mixin(target, ele);
+    return target;
+  }
 
-exports.mixinEmitter = (obj, emitter) => {
-  let proto = emitter.constructor.prototype;
-  for (let key of Object.keys(proto)) {
-    let val = proto[key];
-    if (typeof val === 'function') {
-      exports.define(obj, key, val.bind(emitter));
-    } else {
-      exports.define(obj, key, val);
+  function mixinEmitter(obj, emitter) {
+    let proto = emitter.constructor.prototype;
+    for (let key of Object.keys(proto)) {
+      let val = proto[key];
+      if (typeof val === 'function') {
+        define(obj, key, val.bind(emitter));
+      } else {
+        define(obj, key, val);
+      }
     }
   }
-};
 
-exports.onExit = callback => {
-  const onExit = (quit, code) => {
-    if (called) return;
+  function onExit(callback) {
+    const onExit = (quit, code) => {
+      if (called) return;
 
-    called = true;
-    fns.forEach(fn => fn());
+      called = true;
+      fns.forEach(fn => fn());
 
-    if (quit === true) {
-      process.exit(128 + code);
+      if (quit === true) {
+        exit(128 + code);
+      }
+    };
+
+    if (fns.length === 0) {
+      onOsSignal(osSignals.SIGTERM).then(() => onExit.bind(null, true, 15));
+      onOsSignal(osSignals.SIGINT).then(() => onExit.bind(null, true, 2));
+      onOsSignal(osSignals.EXIT).then((...args) => onExit(...args));
```

Comment:
Failing to pass the arguments `process.once` to `onExit` results in the cursor disappearing in the terminal when running the tests which is super odd and which I haven't understood. 

The arguments were already passed through before this commit so I guess you're aware of the issue, if you can explain why I'd be very interested. 

Even more so because the arguments are: `[0]` which means the code gets passed as first argument which is not what `onExit` expects Oo. Something looks fishy to me :thinking: